apps/dashboard/src/modules/user/components/UserSettingsComponent.vue

```diff
@@ -80,6 +87,58 @@ import { useConfirm } from "primevue/useconfirm";
 import { useToast } from "primevue/usetoast";
 import apiService from "@/services/ApiService";
 import { handleError } from "@/utils/errorUtils";
+import { useUserStore } from "@sudosos/sudosos-frontend-common";
+
+const status: Ref<string> = ref("");
+
+async function startScan() {
+  try {
+    if (!('NDEFReader' in window)) {
+      status.value = "Web NFC is not supported on this device.";
+      console.log('no');
+      toast.add({
+            severity: "error",
+            summary: "Unsupported",
+            detail: "Browser is not mobile and does not have Web NFC.",
+            life: 3000,
+          });
+    } else {
+      const ndef = new (window as any).NDEFReader();
+      await ndef.scan();
+      toast.add({
+        severity: "info",
+        summary: "Scan started",
+        detail: "NFC Scan started, please hold your NFC card to your phone!",
+        life: 3000,
+      });
+
+      // Handle the NFC tag when it is detected
+      ndef.onreading = async (event: NDEFReadingEvent) => {
+
+        // Extract and log the NFC tag's ID (serial number)
+        const tagId = event.serialNumber;
+        toast.add({
+          severity: "success",
+          summary: "Scan finished",
+          detail: "NFC Scan finished, NFC ID added.",
+          life: 3000,
+        });
+        console.log(tagId);
+        await apiService.user.updateUserNfc(useUserStore().getCurrentUser.user.id, { nfcCode: tagId });
+      };
```

Comment:
Try and use the `.then` here as well since you only want to show the 'NFC ID added` if the API call actually worked.

Note that we have a general API error helper which can be found [here](https://github.com/GEWIS/sudosos-frontend/blob/0be9f4395ce5e9be8a169f2a75aad75cc2775f5b/apps/dashboard/src/utils/errorUtils.ts#L17).